components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/common/InboundOneTimeTriggerRequestProcessor.java

```diff
@@ -0,0 +1,112 @@
+
+    public InboundOneTimeTriggerRequestProcessor() {
+        dataStore = InboundEndpointsDataStore.getInstance();
+    }
+
+    /**
+     * Based on the coordination option schedule the task with NTASK or run as a
+     * background thread
+     *
+     * @param task
+     * @param endpointPostfix
+     */
+    protected void start(OneTimeTriggerInboundTask task, String endpointPostfix) {
+        log.info("Starting the inbound endpoint " + name + ", with coordination " + coordination
+                + ". Type : " + endpointPostfix);
+        if (coordination) {
+            try {
+                TaskDescription taskDescription = new TaskDescription();
+                taskDescription.setName(name + "-" + endpointPostfix);
+                taskDescription.setTaskGroup(endpointPostfix);
+                taskDescription.setInterval(TASK_THRESHOLD_INTERVAL);
+                taskDescription.setIntervalInMs(true);
+                taskDescription.addResource(TaskDescription.INSTANCE, task);
+                taskDescription.addResource(TaskDescription.CLASSNAME, task.getClass().getName());
+                startUpController = new StartUpController();
+                startUpController.setTaskDescription(taskDescription);
+                startUpController.init(synapseEnvironment);
+            } catch (Exception e) {
+                log.error("Error starting the inbound endpoint " + name
+                        + ". Unable to schedule the task. " + e.getLocalizedMessage(), e);
+            }
+        } else {
+            PrivilegedCarbonContext carbonContext =
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext();
+            int tenantId = carbonContext.getTenantId();
+            String tenantDomain = null;
+            if (tenantId != MultitenantConstants.SUPER_TENANT_ID) {
+                tenantDomain = carbonContext.getTenantDomain();
+                if (!dataStore.isPollingEndpointRegistered(tenantDomain, name)) {
+                    dataStore.registerPollingingEndpoint(tenantDomain, name);
+                }
+            }
+            inboundRunner = new OneTimeTriggerInboundRunner(task, tenantDomain);
+            runningThread = new Thread(inboundRunner);
+            runningThread.start();
+        }
+    }
+
+    /**
+     * Stop the inbound polling processor This will be called when inbound is
+     * undeployed/redeployed or when server stop
+     */
+    public void destroy() {
+        log.info("Inbound endpoint " + name + " stopping.");
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        int tenantId = carbonContext.getTenantId();
+        if (tenantId != MultitenantConstants.SUPER_TENANT_ID) {
+            dataStore.unregisterPollingEndpoint(carbonContext.getTenantDomain(), name);
+        }
+        if (startUpController != null) {
+            startUpController.destroy();
+        } else if (runningThread != null) {
+            try {
```

Comment:
included to consider case of reconnection, at that thread holds for reconnection.